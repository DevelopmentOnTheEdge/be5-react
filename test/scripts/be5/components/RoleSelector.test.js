import React from 'react';
import renderer from 'react-test-renderer';
import RoleBox, {Role} from '../../../../src/scripts/be5/components/RoleSelector.js';
import be5 from '../../../../src/scripts/be5/be5';
import {shallow, mount} from 'enzyme';


test('render', () => {
    be5.net.request = function (path, attr, callback) {
      callback({"availableRoles":["Administrator", "Manager"],"selectedRoles":["Manager"]})
    };

    const component = renderer.create(
        <RoleBox />
    );
    expect(component.toJSON()).toMatchSnapshot();
});

test('calls', () => {
  const handle = jest.fn();

  be5.net.request = function (path, attr, callback) {
    handle(path, attr, callback);
    callback({"availableRoles":["Administrator", "Manager", "Role1", "Role2"],"selectedRoles":["Manager", "Role1"]})
  };

  const wrapper = mount(
    <RoleBox />
  );
  expect(handle.mock.calls.length).toBe(1);
  expect(handle.mock.calls[0]).toEqual(["roleSelector", {}, expect.any(Function)]);

  expect(wrapper.find('input[type="checkbox"]').length).toEqual(4);
  expect(wrapper.find('#Administrator-checkbox').props().checked).toEqual(false);
  expect(wrapper.find('#Manager-checkbox').props().checked).toEqual(true);

  wrapper.find('#Administrator-checkbox').simulate('change');
  expect(handle.mock.calls[1]).toEqual(
    ["roleSelector/select", {"roles": "Manager,Role1,Administrator"}, expect.any(Function)]);

  wrapper.find('#Manager-checkbox').simulate('change');
  expect(handle.mock.calls[2]).toEqual(
    ["roleSelector/select", {"roles": "Role1"}, expect.any(Function)]);

  wrapper.find('.enable-all').simulate('click');
  expect(handle.mock.calls[3]).toEqual(
    ["roleSelector/select", {"roles": "Administrator,Manager,Role1,Role2"}, expect.any(Function)]);

  wrapper.find('.disable-all').simulate('click');
  expect(handle.mock.calls[4]).toEqual(
    ["roleSelector/select", {"roles": ""}, expect.any(Function)]);

});
