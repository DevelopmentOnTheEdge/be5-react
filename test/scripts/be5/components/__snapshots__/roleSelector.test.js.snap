// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render 1`] = `
<div
  className="roleBox"
>
  <div
    className="dropdown"
    onKeyDown={[Function]}
  >
    <button
      aria-expanded={false}
      aria-haspopup={true}
      className="dropdown-toggle btn btn-secondary"
      onClick={[Function]}
      type="button"
    >
      Roles
    </button>
    <div
      aria-hidden={true}
      className="dropdown-menu"
      role="menu"
      tabIndex="-1"
    >
      <div
        className="role"
      >
        <input
          checked={true}
          id="Guest-checkbox"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="Guest-checkbox"
        >
          <span
            className="checkBox"
          />
          Guest
        </label>
      </div>
      <div
        className="role"
      >
        <input
          checked={false}
          id="Administrator-checkbox"
          onChange={[Function]}
          type="checkbox"
        />
        <label
          htmlFor="Administrator-checkbox"
        >
          <span
            className="checkBox"
          />
          Administrator
        </label>
      </div>
      <div
        className="dropdown-divider"
        onClick={[Function]}
        tabIndex="-1"
        type={undefined}
      />
      <button
        className="dropdown-item"
        onClick={[Function]}
        tabIndex="0"
        type="button"
      >
        кнопки в разработке
        <br />
        <button
          className="btn btn-primary btn-sm"
          onClick={[Function]}
          type="button"
        >
          Выбрать всё
        </button>
         
        <button
          className="btn btn-secondary btn-sm"
          onClick={[Function]}
          type="button"
        >
          Очистить всё
        </button>
      </button>
    </div>
  </div>
</div>
`;
